### PR TITLE
Allow GET requests to history through

### PR DESCRIPTION
**Pull request by phamann**
Ensures that `GET` requests with `X-FT-Session-Token` can pass through to routing.

/cc @adgad @wheresrhys

### CODE REVIEW

**Comment by adgad** [comment]
LGTM -  Fancy adding a quick test in here?

https://github.com/Financial-Times/next-user-preferences-api-v2/blob/master/tests/auth.spec.js

May need to refactor all this authentication logic soon, it's getting quite unwieldy!

**Comment by phamann** [comment]
@adgad have now added tests :smile:

File: server/lib/session-middleware.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ var fetchres = require('fetchres');
 var errors = require('express-errors-handler');
 
 module.exports = function (req, res, next) {
-	var pathGuid = (/\/User\:guid-([-\w]+)/.exec(req.path) || [])[1];
+	var pathGuid = (/\/User\:guid-([-\w]+)|\/history\/([-\w]+)/.exec(req.path) || [])[1];
```
Comment (wheresrhys) [inline server/lib/session-middleware.js]: should've picked up on this earlier, but is it too late to use a `User:guid-` prefix for the history endpoint?

Comment (phamann) [inline server/lib/session-middleware.js]: No i can easily refactor in another PR. But out-of-interest what is the reasoning for this prefix?

Comment (wheresrhys) [inline server/lib/session-middleware.js]: I'm not 100% convinced of its usefulness, but urls were originally all of the form /{object}/{did something to}/{typeof thing}:{thingId}, which is pretty extensible. I think extensibility was the driving reason, though I wasn't really involved at that stage.

**Pull request merged**